foreshadow/column_sharer.py

```diff
@@ -3,46 +3,71 @@
 
 
 class ColumnSharer(MutableMapping):
-    """main cache-class to be used as single-instance to share data."""
+    """Main cache-class to be used as single-instance to share data."""
 
     def __init__(self, *args, **kwargs):
-        super(ColumnSharer, self).__init__(*args, **kwargs)
-        self.store = defaultdict(lambda: {})  # will have a nested dict for
-        # every key, which holds {column: key-column info}
+        super().__init__(*args, **kwargs)
+        self.store = defaultdict(lambda: defaultdict(lambda: None))  # will
+        # have a nested defaultdict for every key, which holds {column:
+        # key-column info} and gives None by default. It is the users
+        # responsibility to make sure returned values are useful.
         acceptable_keys = {"intent": True, "domain": True, "metastat": True}
-        for key in acceptable_keys:
-            self.store[key] = {}
         self.__acceptable_keys = defaultdict(lambda: False, acceptable_keys)
-        self.__registered_keys = defaultdict(lambda: False)
 
     def __getitem__(self, key_list):
         """Override getitem to support multi key accessing simultaneously.
 
+        You can access this data structure in two ways. The first is with
+        the knowledge that the internal implementation uses nested dicts:
+
+        >>> cs = ColumnSharer()
```

Comment:
Can we double check that these show up in the sphinx docs? I think we might need to either put them in a special "code" sphinx environment or in the examples section for them to show up properly.